filters/scheduler/fifo_test.go

```diff
@@ -296,7 +297,7 @@ func TestFifoWithBody(t *testing.T) {
 			rt := routing.New(ro)
 			defer rt.Close()
 			<-rt.FirstLoad()
-			tracer := &testTracer{MockTracer: mocktracer.New()}
+			tracer := &tracingtest.MockTracer{MockTracer: mocktracer.New()}
```

Comment:
Lets check all callsites using `mocktracer.New()` - I think we have multiple various ways to wait for finished spans.
Replacing them with this new tracingtest.NewTracer should simplify tests.

Reply:
Actually, three tracers created and were not used, I have removed them: https://github.com/zalando/skipper/pull/3322/files#diff-703dc4a90e39e2396aa495f14a94d9132ffc097b1be0cb2c01fb244d116289a1L299-L642


The only one left is using FinishedSpans() method of MockTracer in an expected way, no need to change that.

Reply:
After this change was made https://github.com/zalando/skipper/pull/3318#discussion_r1855180957 we can be sure only this FinishedSpans() method is called https://github.com/zalando/skipper/pull/3322/files#diff-3f5e5fd69ecb51b0c383da632e30fe4bd12ffb855d983235b063b70d0a17ac00R30-R45 .